lib/std/atomic.zig

```diff
@@ -408,79 +408,87 @@ test spinLoopHint {
     }
 }
 
+pub fn cacheLineForCpu(cpu: std.Target.Cpu) u16 {
+    return switch (cpu.arch) {
+        // x86_64: Starting from Intel's Sandy Bridge, the spatial prefetcher pulls in pairs of 64-byte cache lines at a time.
+        // - https://www.intel.com/content/dam/www/public/us/en/documents/manuals/64-ia-32-architectures-optimization-manual.pdf
+        // - https://github.com/facebook/folly/blob/1b5288e6eea6df074758f877c849b6e73bbb9fbb/folly/lang/Align.h#L107
+        //
+        // aarch64: Some big.LITTLE ARM archs have "big" cores with 128-byte cache lines:
+        // - https://www.mono-project.com/news/2016/09/12/arm64-icache/
+        // - https://cpufun.substack.com/p/more-m1-fun-hardware-information
+        //
+        // - https://github.com/torvalds/linux/blob/3a7e02c040b130b5545e4b115aada7bacd80a2b6/arch/arc/Kconfig#L212
+        // - https://github.com/golang/go/blob/3dd58676054223962cd915bb0934d1f9f489d4d2/src/internal/cpu/cpu_ppc64x.go#L9
+        .x86_64,
+        .aarch64,
+        .aarch64_be,
+        .arc,
+        .powerpc64,
+        .powerpc64le,
+        => 128,
+
+        // https://github.com/llvm/llvm-project/blob/e379094328e49731a606304f7e3559d4f1fa96f9/clang/lib/Basic/Targets/Hexagon.h#L145-L151
+        .hexagon,
+        => if (std.Target.hexagon.featureSetHas(cpu.features, .v73)) 64 else 32,
+
+        // - https://github.com/golang/go/blob/3dd58676054223962cd915bb0934d1f9f489d4d2/src/internal/cpu/cpu_arm.go#L7
+        // - https://github.com/golang/go/blob/3dd58676054223962cd915bb0934d1f9f489d4d2/src/internal/cpu/cpu_mips.go#L7
+        // - https://github.com/golang/go/blob/3dd58676054223962cd915bb0934d1f9f489d4d2/src/internal/cpu/cpu_mipsle.go#L7
+        // - https://github.com/golang/go/blob/3dd58676054223962cd915bb0934d1f9f489d4d2/src/internal/cpu/cpu_mips64x.go#L9
+        // - https://github.com/golang/go/blob/3dd58676054223962cd915bb0934d1f9f489d4d2/src/internal/cpu/cpu_riscv64.go#L7
```

Comment:
This link is outdated: https://github.com/golang/go/commit/5e31f78c8a4ed1b872ddc194f0cd1ae931b37d7e.
Back when I was doing the fence PR, I was kinda puzzled about this number, but finally got reminded to double check now.

Reply:
We should probably just change most of these to refer to the Linux kernel anyway; that's usually a better source for this information.

Reply:
I agree there.